time-it-builder/src/lib.rs

```diff
@@ -0,0 +1,61 @@
+use lazy_init::Lazy;
+use parking_lot::RwLock;
+use std::{sync::Arc, time::Duration};
+
+lazy_static::lazy_static! {
+    static ref TIMER_CONFIG: RwLock<TimeItConfig> = RwLock::new(TimeItConfig::default());
+    static ref LAZY_TIMER_CONFIG: Lazy<Arc<TimeItConfig>> = Lazy::new();
+}
+
+#[derive(Clone)]
+pub struct TimeItConfig {
+    pub action: Arc<Box<dyn Fn(Duration) -> () + Send + Sync>>,
+}
+
+impl Default for TimeItConfig {
+    fn default() -> Self {
+        TimeItConfig {
+            action: Arc::new(Box::new(|duration| {
+                let time_elapsed = duration.as_millis();
+                println!("[TimeIt] Time elapsed: {}ms", time_elapsed);
+            })),
+        }
+    }
+}
+
+pub struct TimeItBuilder {}
+
+impl TimeItBuilder {
+    pub fn new() -> Self {
+        TimeItBuilder {}
+    }
+
+    /// This function is called when the timer is done.
+    /// The first argument is the name of the timer, the second is the duration.
+    pub fn time_it(self, action: Box<dyn Fn(Duration) -> () + Send + Sync>) -> Self {
```

Comment:
```suggestion
    pub fn time_it(&self, action: Box<dyn Fn(Duration) -> () + Send + Sync>) -> &Self {
```
should make these references. I made the mistake of not doing refernces on a builder in the rust hsc and it lead to lines of code like this some times, if a user wanted to assign just one value, for example in a conditional
```
let builder = builder.something("hi");
```